### PR TITLE
[CPDLP-1335] Internal API NPQ funding

### PR DESCRIPTION
**Pull request by asmega**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1335

### Changes proposed in this pull request

- Added service class to calculate NPQ funding eligibility based on TRN and course
- Added private api endpoint so it is accessible via NPQ rails app api call

### Guidance to review

- Create a private api token
- Using this token request NPQ funding with `trn` and `npq_course_identifier`
- Should return appropriate result

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -52,6 +52,9 @@
 
     namespace :v1 do
       concern :participant_actions, Participants::Routing.new
+
+      resources :npq_funding, only: [:index], path: "npq-funding"
```
Comment (Lockyy) [inline config/routes.rb:56]: Is this more of a show? We're looking up data on a single TRN and getting one hash of data back.

**Review by Lockyy — APPROVED** [review]
LGTM

File: config/routes.rb (hunk 1 above)
Comment (asmega) [reply config/routes.rb]: sure, I've changed this to a `show` now

**Review by Lockyy — APPROVED** [review]
Approving, Phil has let me know that the end-to-end failures are unrelated and being fixed by Matt

**Pull request merged**